crates/ty_project/src/metadata/options.rs

```diff
@@ -277,14 +278,36 @@ impl Options {
             roots
         };
 
-        let settings = SearchPathSettings {
-            extra_paths: environment
+        // collect the existing site packages
+        let mut extra_paths: Vec<SystemPathBuf> = Vec::new();
+
+        // read all the paths off the PYTHONPATH environment variable, check
+        // they exists, and add them to the vec of extra_paths as they should
+        // be checked before site-packages just like python interpreter does
+        if let Ok(python_path) = env::var("PYTHONPATH") {
+            let splits: Vec<SystemPathBuf> = python_path
+                .split(':')
+                .filter_map(|str_path| {
+                    let possible_path = std::path::PathBuf::from(str_path);
+                    match possible_path.try_exists() {
+                        Ok(true) => SystemPathBuf::from_path_buf(possible_path).ok(),
+                        _ => None,
+                    }
+                })
+                .collect();
+            extra_paths.extend(splits);
+        }
+        extra_paths.extend(
+            environment
                 .extra_paths
                 .as_deref()
                 .unwrap_or_default()
                 .iter()
-                .map(|path| path.absolute(project_root, system))
-                .collect(),
+                .map(|path| path.absolute(project_root, system)),
+        );
+
+        let settings = SearchPathSettings {
```

Comment:
@AlexWaygood any thoughts on whether we should move this logic into `SearchPaths::from_settings`?

Reply:
No strong opinion really!